localstack/services/sns/provider.py

```diff
@@ -358,21 +360,23 @@ def create_platform_endpoint(
 
     def unsubscribe(self, context: RequestContext, subscription_arn: subscriptionARN) -> None:
         count = len(subscription_arn.split(":"))
+        if count < 6:
+            raise InvalidParameterException(
+                f"Invalid parameter: SubscriptionArn Reason: An ARN must have at least 6 elements, not {count}"
+            )
         try:
             parsed_arn = parse_arn(subscription_arn)
         except InvalidArnException:
             # TODO: check for invalid SubscriptionGUID
-            if count < 6:
-                raise InvalidParameterException(
-                    f"Invalid parameter: SubscriptionArn Reason: An ARN must have at least 6 elements, not {count}"
-                )
+            raise InvalidParameterException(
+                f"Invalid parameter: SubscriptionArn Reason: An ARN must have at least 6 elements, not {count}"
+            )
```

Comment:
It seems a bit weird to have this code twice. Is there a particular reason why we check the count ourselves and then let moto do the conversion with the same check again?

Reply:
Right yes just checked the botocore code, and they also validate the `6 elements` part. Will rewrite and update the test like Alex said at the same time. Thanks a lot! 👍